File: AFMobile/AFWinPhone/AFWinPhone/builders/widgets/AbstractWidgetBuilder.cs
using AFWinPhone.components.parts;
using System;
using Windows.UI.Xaml;

namespace AFWinPhone.builders.widgets
{
    /// <summary>
    /// Template for widget builders and it defines methods which every widget builder should have.
    /// </summary>
    public interface AbstractWidgetBuilder
    {

        /// <summary>
        /// This method is used for build graphical representation of widget - field's active element,
        /// that means without validation or label elements.
        /// </summary>
        /// <returns>graphical representation of field's active element</returns>
        FrameworkElement buildFieldView();

        /// <summary>
        /// Sets data to widget - field's active element
        /// </summary>
        /// <param name="value">data to set (usually String)</param>
        void setData(Object value);

        /// <summary>
        /// Gets data from widget - field's active element
        /// </summary>
        /// <returns>data from widget</returns>
        Object getData();
    }
}

File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/widgets/AbstractWidgetBuilder.java
package cz.cvut.fel.matyapav.afandroid.builders.widgets;

import android.app.Activity;
import android.view.View;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;

/**
 * Template for widget builders and it defines methods which every widget builder should have.
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz)
 *
 * @since 1.0.0.
 */
public interface AbstractWidgetBuilder {

    /**
     * This method is used for build graphical representation of widget - field's active element,
     * that means without validation or label elements.
     * @param activity is needed to be able to dynamically create a component
     * @return graphical representation of field's active element
     */
    View buildFieldView(Activity activity);

    /**
     * Sets data to widget - field's active element
     * @param value data to set (usually String)
     */
    void setData(Object value);

    /**
     * Gets data from widget - field's active element
     * @return data from widget
     */
    Object getData();
}
